companion/src/instruction/bristol.rs

```diff
@@ -10,21 +10,102 @@ use super::Parser;
 pub struct InsParser {
     line: String,
     reader: BufReader<File>,
+    pub n_gate: usize,
```

Comment:
This is a nitpick, but it might make sense to abstract these fields as a `BristolHeader` and `impl FromStr` for that header, since that gives us `parse()` and sensible error handling.

Reply:
I appreciate the feedback! This is exactly the kind of nitpick I'm looking for. If the header parser was written in Python I'd consider it unacceptably ugly, but I'm not quite familiar enough with Rust to know how to make it better.

Reply:
Converted this to use FromStr. A bit iffy on the error handling though - there are a lot of `unwrap`'s in there that seem like they would be better suited to using the `?` syntax, but rustc doesn't like it because the errors are different types.

Reply:
Yeah, one of the annoying things about `?` is that it can't implicitly convert errors without proper `From` or `Into` traits between them.